pkg/cloud/cloud.go

```diff
@@ -124,6 +126,27 @@ func (f *providerFactory) Get(config *configv1alpha1.SubmarinerConfig, eventsRec
 		return nil, true, err
 	}
 
+	// Check for CloudOpts set as annotations in SubmarinerConfig
+	annotations := config.Annotations
+	if annotations != nil {
+		if vpcID, exists := annotations["submariner.io/vpc-id"]; exists {
+			info.CloudOptions = append(info.CloudOptions, cpaws.WithVPCName(vpcID))
+		}
+		if subnetIDList, exists := annotations["submariner.io/subnet-id-list"]; exists {
+			subnetIDs := strings.Split(subnetIDList, ",")
+			for i := range subnetIDs {
+				subnetIDs[i] = strings.TrimSpace(subnetIDs[i])
+			}
+			info.CloudOptions = append(info.CloudOptions, cpaws.WithPublicSubnetList(subnetIDs))
+		}
+		if controlPlaneSGID, exists := annotations["submariner.io/control-plane-sg-id"]; exists {
+			info.CloudOptions = append(info.CloudOptions, cpaws.WithControlPlaneSecurityGroup(controlPlaneSGID))
+		}
+		if workerSGID, exists := annotations["submariner.io/worker-sg-id"]; exists {
+			info.CloudOptions = append(info.CloudOptions, cpaws.WithWorkerSecurityGroup(workerSGID))
+		}
+	}
```

Comment:
This really belongs in the AWS cloud code. I think we should store the `SubmarinerConfig` in the `provider.Info` struct rather than the AWS-specific `CloudOptions`.

Reply:
@tpantelis Shouldn't we be moving to SubmarinerConfig variables inside submariner config itself , is this ok to have as an interim fix?

Reply:
@aswinsuryan Cloud specific variables don't make sense in Submariner CR as submariner CR doesn't do anything with cloud-prepare.

We may have scenarios where we need similar CloudOptions for other clouds too. For now we can leave this here, but can leave a TODO to refactor it later.

Reply:
I’d rather not leave a TODO - in my experience they rarely get addressed later. It seems like we’re taking about 2 different things here. One is how the user specifies the cloud options - right now it’s via annotations on the SC CR. It could be via fields. I’m not sure how else we’d do it. The fact is whatever we do here likely won’t be changed later as that would require migration, docs change etc. So let’s decide the best way now - annotations seem reasonable and the last east invasive. The second is how we handle it in code. I’m merely the suggesting we move the parsing into the AWS code where it belongs by passing the SC CR to the provider instance.  This is simple to do. I can make the change.

Reply:
Done.